sources/web/datalab/polymer/components/datalab-app/datalab-app.ts

```diff
@@ -55,6 +55,14 @@ class DatalabAppElement extends Polymer.Element {
 
     // Will be called after the custom element is done rendering.
     window.addEventListener('WebComponentsReady', this._boundResizeHandler, true);
+
+    ApiManager.disconnectedHandler = () => {
```

Comment:
I like this better than the previous iteration. Adding the isConnected status in ApiManager is also an improvement.

To clarify the comment I made in Utils.ts about connect clearing the toast, as far as I can see everything will work fine for now with the current code because you are using the toast only for the connection status. I am concerned that an issue may arise later when you start using the toast for other things besides connection status. The essence of my concern is, what if a toast is posted about something other than connection status, but then immediately after that happens, a connection is reestablished; then the toast about that other thing will be cleared. But I think we can probably wait on that until it happens, if ever.